metaconfig-schema/src/main/scala/metaconfig/schema/Schema.scala

```diff
@@ -0,0 +1,85 @@
+package metaconfig.schema
+
+import metaconfig.generic.Setting
+import metaconfig.generic.Settings
+import ujson._
+
+object Schema {
+
+  def schema[T <: Product](
```

Comment:
How about we use `T:ConfEncoder` instead? The `docs` module uses `Product` which is kind of a hack which should be replaced with `ConfEncoder`